src/main/java/com/openlattice/datastore/permissions/controllers/PermissionsController.java

```diff
@@ -64,62 +60,60 @@
             method = RequestMethod.PATCH,
             consumes = MediaType.APPLICATION_JSON_VALUE )
     public Void updateAcl( @RequestBody AclData req ) {
+        return updateAcls( ImmutableList.of( req ) );
+    }
+
+    @Override
+    @Timed
+    @RequestMapping(
+            path = { UPDATE },
+            method = RequestMethod.PATCH,
+            consumes = MediaType.APPLICATION_JSON_VALUE )
+    public Void updateAcls( @RequestBody List<AclData> req ) {
+
+        Map<Action, List<Acl>> requestsByActionType = req.stream().collect( Collectors
+                .groupingBy( AclData::getAction, Collectors.mapping( AclData::getAcl, Collectors.toList() ) ) );
+
         /*
          * Ensure that the user has alter permissions on Acl permissions being modified
          */
-        final Acl acl = req.getAcl();
-        final AclKey aclKeys = new AclKey( acl.getAclKey() );
-        if ( isAuthorized( Permission.OWNER ).test( aclKeys ) ) {
-            switch ( req.getAction() ) {
+        Set<AclKey> unauthorizedAclKeys = authorizations
+                .accessChecksForPrincipals( requestsByActionType.entrySet().stream()
+                        .filter( entry -> !entry.getKey().equals( Action.REQUEST ) )
+                        .flatMap( entry -> entry.getValue().stream() )
+                        .map( acl -> new AccessCheck( new AclKey( acl.getAclKey() ), EnumSet.of( Permission.OWNER ) ) )
+                        .collect( Collectors.toSet() ), Principals.getCurrentPrincipals() )
+                .filter( authorization -> !authorization.getPermissions().get( Permission.OWNER ) )
+                .map( Authorization::getAclKey )
+                .collect( Collectors.toSet() );
+
+        if ( !unauthorizedAclKeys.isEmpty() ) {
+            throw new ForbiddenException( "Only owner of securable objects " + unauthorizedAclKeys +
+                    " can access other users' access rights." );
+        }
+
+        requestsByActionType.entrySet().forEach( entry -> {
```

Comment:
You could also use Map.foreach { (key, value) -> ...}

Reply:
yeah that's way cleaner :)